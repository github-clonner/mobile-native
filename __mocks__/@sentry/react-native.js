export const captureMessage = jest.fn();
export default {};
