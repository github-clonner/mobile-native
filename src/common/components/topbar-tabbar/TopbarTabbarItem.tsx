import React, { Component } from 'react';

import { inject, observer } from 'mobx-react';

import {
  Text,
  TouchableOpacity,
  StyleSheet,
  View,
  StyleProp,
  ViewStyle,
  GestureResponderEvent,
} from 'react-native';

import { CommonStyle } from '../../../styles/Common';
import colors from '../../../styles/Colors';
import ThemedStyles from '../../../styles/ThemedStyles';
import { useTopbarBarStore } from './TopbarTabbarContext';

interface Props {
  id: string;
  children: string;
  //onPress: (e: GestureResponderEvent) => {};
  style?: StyleProp<ViewStyle>;
}

/**
 * Topbar Tabbar Item
 */
export const TopbarTabbarItem = observer((props: Props) => {
  const store = useTopbarBarStore();

  const isSelected = (): ViewStyle | null => {
    // const activeTabId = store.activeTabId;
    // return activeTabId == txt ? styles.tabSelected : null;
    return store.activeTabId === props.id ? styles.tabSelected : null;
  };

  const onPress = (e: GestureResponderEvent) => {
    store.setActiveTabId(props.id);
  };

  return (
    <TouchableOpacity
      style={[
        styles.tab,
        {
          borderBottomColor: ThemedStyles.getColor('secondary_background'),
        },
        isSelected(),
      ]}
      onPress={onPress}>
      <Text style={CommonStyle.fontM}>{props.children}</Text>
    </TouchableOpacity>
  );
});

export default TopbarTabbarItem;

//TODO: move to common style
const styles = StyleSheet.create({
  tab: {
    justifyContent: 'center',
    alignItems: 'center',
    borderBottomWidth: 4,
    paddingHorizontal: 0,
    marginRight: 20,
  },
  tabSelected: {
    borderBottomColor: colors.primary,
  },
});
