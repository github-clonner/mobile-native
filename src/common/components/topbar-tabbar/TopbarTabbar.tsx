import React, { Component, ReactChildren, Children, useEffect } from 'react';

import { inject, observer, useLocalStore } from 'mobx-react';

import { Text, TouchableOpacity, StyleSheet, View } from 'react-native';
import ThemedStyles from '../../../styles/ThemedStyles';
import {
  useTopbarBarStore,
  storesContext,
  createStore,
} from './TopbarTabbarContext';
import type TopbarTabbarItem from './TopbarTabbarItem';

interface Props {
  activeTabId: string;
  onTabChange: (tabId: string) => void;
  children: typeof TopbarTabbarItem[] | any;
}

/**
 * Topbar Tabbar
 */
export const TopbarTabbar = observer((props: Props) => {
  const store = useLocalStore(createStore);

  useEffect(() => {
    store.setActiveTabId(props.activeTabId);
  }, [props.activeTabId]);

  useEffect(() => {
    if (store.activeTabId && props.activeTabId !== store.activeTabId)
      props.onTabChange(store.activeTabId);
  }, [store.activeTabId]);

  return (
    <View>
      <storesContext.Provider value={store}>
        <View
          style={[
            styles.container,
            ThemedStyles.style.borderBottomHair,
            ThemedStyles.style.borderPrimary,
            ThemedStyles.style.backgroundSecondary,
          ]}>
          <View style={styles.topbar}>{props.children}</View>
        </View>
      </storesContext.Provider>
    </View>
  );
});

export default TopbarTabbar;

//TODO: move to common style
const styles = StyleSheet.create({
  container: {
    height: 42,
    display: 'flex',
    flexDirection: 'row',
    paddingTop: 0,
  },
  topbar: {
    flex: 1,
    justifyContent: 'flex-start',
    flexDirection: 'row',
    paddingLeft: 20,
    paddingRight: 20,
  },
  button: {
    padding: 8,
  },
});
