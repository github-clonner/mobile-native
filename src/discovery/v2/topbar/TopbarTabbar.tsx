import React, { Component } from 'react';

import { inject, observer } from 'mobx-react';

import { Text, TouchableOpacity, StyleSheet, View } from 'react-native';

import { CommonStyle } from '../../../styles/Common';
import colors from '../../../styles/Colors';
import i18n from '../../../common/services/i18n.service';
import ThemedStyles from '../../../styles/ThemedStyles';
import DiscoveryV2Store from '../DiscoveryV2Store';
import { useStores } from '../../../common/hooks/use-stores';

interface Props {}

/**
 * Newsfeed top bar
 */
export const TopbarTabbar = observer(() => {
  const { discoveryV2 } = useStores();

  /**
   * Selected
   * @param {string} txt
   */
  const selected = (txt) => {
    const activeTabId = discoveryV2.activeTabId;
    return activeTabId == txt ? styles.tabSelected : null;
  };

  /**
   * Render
   */

  return (
    <View>
      <View
        style={[
          styles.container,
          ThemedStyles.style.borderBottomHair,
          ThemedStyles.style.borderPrimary,
          ThemedStyles.style.backgroundSecondary,
        ]}>
        <View style={styles.topbar}>
          <TouchableOpacity
            style={[
              styles.tab,
              {
                borderBottomColor: ThemedStyles.getColor(
                  'secondary_background',
                ),
              },
              selected('foryou'),
            ]}
            onPress={() => discoveryV2.setTabId('foryou')}>
            <Text style={CommonStyle.fontM}>Just for you</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={[
              styles.tab,
              {
                borderBottomColor: ThemedStyles.getColor(
                  'secondary_background',
                ),
              },
              selected('tags'),
            ]}
            onPress={() => discoveryV2.setTabId('tags')}>
            <Text style={CommonStyle.fontM}>Discovery by Tags</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
});

//TODO: move to common style
const styles = StyleSheet.create({
  container: {
    height: 42,
    display: 'flex',
    flexDirection: 'row',
    paddingTop: 6,
    // borderBottomWidth: StyleSheet.hairlineWidth,
    // borderBottomColor: '#EEE',
  },
  topbar: {
    flex: 1,
    justifyContent: 'flex-start',
    flexDirection: 'row',
    paddingLeft: 16,
    paddingRight: 16,
  },
  button: {
    padding: 8,
  },
  tab: {
    justifyContent: 'center',
    alignItems: 'center',
    borderBottomWidth: 4,
    paddingHorizontal: 0,
    marginRight: 20,
  },
  tabSelected: {
    borderBottomColor: colors.primary,
  },
});
