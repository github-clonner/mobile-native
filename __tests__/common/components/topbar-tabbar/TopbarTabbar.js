import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { shallow, render } from 'enzyme';
import TopbarTabbar from '../../../../src/common/components/topbar-tabbar/TopbarTabbar';
import TopbarTabbarItem from '../../../../src/common/components/topbar-tabbar/TopbarTabbarItem';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const topbarTabbar = renderer.create(<TopbarTabbar></TopbarTabbar>).toJSON();
  expect(topbarTabbar).toMatchSnapshot();
});

it('renders with a tabitem correctly', () => {
  const topbarTabbar = renderer
    .create(
      <TopbarTabbar>
        <TopbarTabbarItem id="first">First Label</TopbarTabbarItem>
      </TopbarTabbar>,
    )
    .toJSON();

  expect(topbarTabbar.children.length).toEqual(1);
  expect(topbarTabbar.children[0]).toBeTruthy();
});
