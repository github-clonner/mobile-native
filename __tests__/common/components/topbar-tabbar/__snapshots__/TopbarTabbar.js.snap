// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "height": 42,
          "paddingTop": 0,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
    />
  </View>
</View>
`;
